.github/workflows/test-plugins-against-cockpit.yml

```diff
@@ -0,0 +1,188 @@
+            fi
+          else
+            echo "No cache file found for ${{ env.TAG }} with plugins version ${{ env.PLUGINS_VERSION }}"
+          fi
+
+      - name: Skip if already tested
+        if: ${{ env.skip_job == 'true' }}
+        run: echo "Skipping job for shell version ${{ env.VERSION }} and plugins version ${{ env.PLUGINS_VERSION }}"
+
+      - name: Setup Node.js
+        if: ${{ env.skip_job != 'true' }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'npm'
+
+      - name: Install dependencies
+        if: ${{ env.skip_job != 'true' }}
+        run: npm ci
+
+      - name: Download build artifact
+        if: ${{ env.skip_job != 'true' }}
+        uses: actions/download-artifact@v4
+        with:
+          name: build
+          path: dist/apps/sag-pkg-community-plugins/
+
+      - name: Get shell app of particular version
+        if: ${{ env.skip_job != 'true' }}
+        uses: SoftwareAG/plugins-e2e-setup/get-shell-app@main
+        with:
+          shell-name: cockpit
+          shell-version: ${{ env.VERSION }}
+          shell-path: dist/apps
+
+      - name: Cypress run
+        if: ${{ env.skip_job != 'true' }}
+        uses: cypress-io/github-action@v5
+        with:
+          start: npm run cypress:ctrl
+          install: false
+          wait-on: 'http://localhost:4200/apps/cockpit/index.html?remotes=%7B"sag-pkg-community-plugins"%3A%5B"ExampleWidgetPluginModule"%2C"DatapointsGraphWidgetModule"%5D%7D#'
+          browser: chrome
+          record: false
+          config-file: cypress.config.ts
+          env: C8Y_CTRL_MODE=mocking,C8Y_CTRL_PROVIDER_VERSION=${{ env.MAJOR }}
+
+      - name: Upload cypress screenshots
+        if: failure() && env.skip_job != 'true'
+        uses: actions/upload-artifact@v4
+        with:
+          retention-days: 5
+          name: cypress-screenshots
+          path: cypress/screenshots
+
+      - name: Upload cypress videos
+        if: always() && env.skip_job != 'true'
+        uses: actions/upload-artifact@v4
+        with:
+          retention-days: 5
+          name: cypress-videos
+          path: cypress/videos
+
+      - name: Create or update JSON file
+        if: ${{ env.skip_job != 'true' && github.event_name == 'schedule' }}
+        run: |
+          file="./cache-files/${{ env.TAG }}-${{ env.PLUGINS_VERSION }}.json"
+          now=$(date --utc +"%Y-%m-%dT%H:%M:%SZ")
+          jq -n --arg shellVersion "${{ env.VERSION }}" --arg pluginsVersion "${{ env.PLUGINS_VERSION }}" --arg lastSuccess "$now" '{shellVersion: $shellVersion, pluginsVersion: $pluginsVersion, lastSuccess: $lastSuccess}' > $file
+          cat $file
+
+      - name: Upload updated JSON file to cache
+        if: ${{ env.skip_job != 'true' && github.event_name == 'schedule' }}
+        uses: actions/cache@v4
+        with:
+          path: ./cache-files
+          key: cache-json-files-${{ runner.os }}-${{ env.TAG }}-${{ env.PLUGINS_VERSION }}
```

Comment:
You might run into concurrency issues as the matrix will run multiple jobs in parallel, overriding a previously uploaded cache.. 
Could be avoided by:
- setting `max-parallel: 1` for the strategy
- or having a separate job, which runs after all the matrix jobs to upload the results.

Reply:
I was indeed having problems with cache overriding by parallel jobs, that's why key contains- it's not possible to have two parallel jobs for one tag (and by 'impossible' I mean it- when I get dist tags from npm, I get object with dist tags as keys, so dist tags are unique in result from collect-shell-versions job, therefore key of cache file is unique).
what scenario are you thinking of that might break workflow with overriding a previously uploaded cache?